[nvbugs/5268808][fix] Fix the potential out-of-range-access issue of allreduce workspace.

hyukn opened the pull request:
This issue is only found for tp=ep=8 on multi-node machine. This could be caused by the race between different tp that triggers the illegal list access.

hyukn commented:
/bot run --only-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #4579](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4579/) [ run ] triggered by Bot

qiaoxj07 commented:
@hyukn  I tired this fix, there will be illegal memory access error on GB200 TEP8 after apply the patch, could you have a look?

tensorrt-cicd commented:
[PR_Github #4579](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4579/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #3295 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3295/) completed with status: 'FAILURE'

hyukn commented:
> @hyukn I tried this fix, there will be illegal memory access error on GB200 TEP8 after apply the patch, could you have a look?

Hi Xianjie. You mean if testing it without multi-node, the test for deepseek on single node will trigger illegal memory access issue?

hyukn committed:
Fix the potential out-of-range-access issue of allreduce workspace.
This issue is only found for tp=ep=8 on multi-node machine. This could be caused by the race between different tp that triggers the illegal list access.
* Reform the workspace allocation strategy to list out of range issue.
* Disable min_latency_mode to avoid moe_allreduce illegal memory access issue under multi-node scenario.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

hyukn commented:
/bot run --only-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #4629](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4629/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4629](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4629/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #3338 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3338/) completed with status: 'FAILURE'

hyukn commented:
/bot run --only-multi-gpu-test

tensorrt-cicd commented:
[PR_Github #4650](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4650/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4673](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4673/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #4673](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4673/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #3370 `(Partly Tested)`](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/3370/) completed with status: 'SUCCESS'

hyukn commented:
/bot run

tensorrt-cicd commented:
[PR_Github #4917](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/4917/) [ run ] triggered by Bot
